### PR TITLE
Component/4226 bottom sheet

### PR DESCRIPTION
**Pull request by Cal-L**
**Description**

#4226 
image

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_
_2. What is the improvement/solution?_

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/component-library/components/BottomSheet/BottomSheet.tsx [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    };
+
+    const gestureHandler = useAnimatedGestureHandler<
+      PanGestureHandlerGestureEvent,
+      { startY: number }
+    >({
+      onStart: (_, ctx) => {
+        ctx.startY = yOffset.value;
+      },
+      onActive: (event, ctx) => {
```
Comment (brianacnguyen) [inline app/component-library/components/BottomSheet/BottomSheet.tsx:83]: should these calculation logic be extracted into a util function to be reused in the future for similar gestures?

File: app/component-library/components/BottomSheet/BottomSheet.tsx [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    });
+
+    const hide = () => {
+      yOffset.value = withTiming(
+        sheetHeight.value,
+        { duration: MANUAL_ANIMATION_DURATION },
+        () => runOnJS(onHide)(),
+      );
+    };
+
```
Comment (brianacnguyen) [inline app/component-library/components/BottomSheet/BottomSheet.tsx:133]: should we add a show?

**Review by brianacnguyen — CHANGES_REQUESTED** [review]
Left quite a few comments

File: app/component-library/components/BottomSheet/BottomSheet.tsx (hunk 1 above)
Comment (Cal-L) [reply app/component-library/components/BottomSheet/BottomSheet.tsx]: May abstract if logic is needed in other components. For example Modal

**Review by brianacnguyen — APPROVED** [review]

**Review by owencraston — APPROVED** [review]
This code is so clean it cleansed my soul 👏  🧼

**Pull request merged**